xwiki-platform-core/xwiki-platform-instance/src/main/java/org/xwiki/instance/script/InstanceScriptService.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.instance.script;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.instance.InstanceId;
+import org.xwiki.instance.InstanceIdManager;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provides instance-specific scripting APIs.
+ * 
+ * @version $Id$
+ * @since 8.3-RC1
+ */
+@Component
+@Named("instance")
+@Singleton
+@Unstable
+public class InstanceScriptService implements ScriptService
+{
+    /**
+     * The key under which the last encountered error is stored in the current execution context.
+     */
+    public static final String INSTANCE_ERROR_KEY = "scriptservice.instance.error";
```

Comment:
This is not needed nor used below and it's no longer a best practice since scripts should now not throw exceptions. See http://markmail.org/message/e77r6zn45teghrmz

Now, I just noticed that I forgot to update http://extensions.xwiki.org/xwiki/bin/view/Extension/Script+Module#HBestPractices which I'll do and close the thread.